<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.335">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Solomon Chak">
<meta name="dcterms.date" content="2023-04-17">

<title>Bioinformatics Class Project Day 2</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="Project_Day2_files/libs/clipboard/clipboard.min.js"></script>
<script src="Project_Day2_files/libs/quarto-html/quarto.js"></script>
<script src="Project_Day2_files/libs/quarto-html/popper.min.js"></script>
<script src="Project_Day2_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="Project_Day2_files/libs/quarto-html/anchor.min.js"></script>
<link href="Project_Day2_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="Project_Day2_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="Project_Day2_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="Project_Day2_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="Project_Day2_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body>

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction">Introduction</a></li>
  <li><a href="#load-libraries" id="toc-load-libraries" class="nav-link" data-scroll-target="#load-libraries">Load libraries</a></li>
  <li><a href="#de-analysis" id="toc-de-analysis" class="nav-link" data-scroll-target="#de-analysis">DE analysis</a></li>
  <li><a href="#rm-analysis" id="toc-rm-analysis" class="nav-link" data-scroll-target="#rm-analysis">RM analysis</a></li>
  </ul>
</nav>
</div>
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Bioinformatics Class Project Day 2</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Solomon Chak </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">April 17, 2023</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<section id="introduction" class="level2">
<h2 class="anchored" data-anchor-id="introduction">Introduction</h2>
<p>Today, we will explore the differential expression (DE) analysis and the repeat masker (RM) analysis based on the assembled transcriptome.</p>
<ul>
<li><p>Last time, you used Galaxy to analyzed the RNA-seq data from three queens and three workers to find differential expressed genes.</p></li>
<li><p>Using the same RNA-seq data, I assembled the transcriptome using a program called Trinity. Then I ran Repeat Masker to identify transposable elements (TEs) in the transcriptome.</p></li>
</ul>
</section>
<section id="load-libraries" class="level2">
<h2 class="anchored" data-anchor-id="load-libraries">Load libraries</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"tidyr"</span>)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"dplyr"</span>)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"ggpubr"</span>)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"stringr"</span>)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"plyr"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="de-analysis" class="level2">
<h2 class="anchored" data-anchor-id="de-analysis">DE analysis</h2>
<p>Now that the differential expression analysis is finished on Galaxy. Download the result and examine it in R.</p>
<p>Let’s first load the table stored in <code>input.matrix.Q_vs_W.DESeq2.txt</code>. Upload this file to posit.cloud.</p>
<p>This file has columns separated by tabs. So we will use <code>read.delim()</code> instead of <code>read.csv().</code></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>df <span class="ot">=</span> <span class="fu">read.delim</span>(<span class="at">file =</span> <span class="st">"input.matrix.Q_vs_W.DESeq2.txt"</span>, <span class="at">row.names =</span> <span class="cn">NULL</span>)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="co"># rename the first column</span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(df)[<span class="dv">1</span>] <span class="ot">=</span> <span class="st">"seqid"</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Preview the file</span></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<hr>
<p>Now, let’s keep only rows (transcripts) that have significant <code>padj</code> <strong>(</strong>adjusted p-values) and have the absolute value of <code>log2FoldChange</code> larger than 10.</p>
<p>Note that fold change is calculated based on sampleA vs.&nbsp;sampleB.</p>
<ul>
<li><p>It’s positive when baseMeanA &gt; baseMeanB.</p></li>
<li><p>It’s negative when baseMeanA &lt; baseMeanB.</p></li>
</ul>
<p>Also note that the adjusted p-value is a test of whether the log2 fold change is <u>different from 0</u>. It is statistically different from testing whether log2 fold change &gt; 10 or &lt;-10. A different test could be used for that but for here, we are doing a crude post-hoc filtering.</p>
<p>Write some codes to make an object called <code>df_de</code> that keep rows from <code>df</code> where <code>padj</code> &lt; 0.05 &amp; abs(<code>log2FoldChange</code>) &gt; 10.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># your code</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">

</div>
<p>Using <code>df_de</code>, write some codes to check how many transcripts are up-regulated in queens and how many transcripts are up-regulated in workers. (Remember that all transcripts in <code>df_de</code> is statistically significantly differentially expressed because we have filtered it by <code>padj</code> already.)</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># your code</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">

</div>
<p>We can do a MA plot. But since we didn’t run DESeq2 in R, we will rely on the package <code>ggpubr</code> to do so.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggmaplot</span>(df, <span class="at">size =</span> <span class="fl">0.5</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Project_Day2_files/figure-html/unnamed-chunk-7-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># size is point size</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<hr>
<p>To explore what genes are being up/down regulated in queens (vs.&nbsp;workers), we can look at the identity of some of the most up/down regulated genes. For illustration purpose, we will only look at the top one.</p>
<p>Because the shrimp transcriptome is not annotated, we will need to:</p>
<ol type="1">
<li><p>retrieve the seqid,</p></li>
<li><p>get the fasta sequence of the transcript, and</p></li>
<li><p>do a blast to find out what this transcript (expressed gene) could be.</p></li>
</ol>
<p>(1) We can find the <code>seqid</code> of the most up-regulated gene and most down-regulated in queens in R.</p>
<p>Remember that you can sort a dataframe with <code>arrange()</code>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Sorting to show the lowest log2FoldChange. </span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Is this up or down regulated?</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="fu">arrange</span>(df_de, log2FoldChange) <span class="sc">%&gt;%</span> <span class="fu">head</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                seqid sampleA sampleB baseMeanA baseMeanB baseMean
1  comp180383_c1_seq4       Q       W 0.0000000  699.5098 349.7549
2  comp181460_c0_seq1       Q       W 0.5026212 1010.7706 505.6366
3  comp176503_c0_seq6       Q       W 0.0000000  309.1014 154.5507
4 comp176628_c1_seq17       Q       W 0.0000000  301.8550 150.9275
5 comp173370_c0_seq13       Q       W 0.0000000  278.4993 139.2496
6  comp179191_c2_seq8       Q       W 0.0000000  266.9005 133.4503
  log2FoldChange    lfcSE      stat       pvalue         padj
1      -11.62299 3.123961 -3.720592 1.987562e-04 2.784806e-02
2      -11.15852 2.733697 -4.081843 4.467989e-05 8.772759e-03
3      -10.44506 1.714528 -6.092089 1.114469e-09 1.720196e-06
4      -10.40991 1.408855 -7.388915 1.480319e-13 9.134437e-10
5      -10.29197 1.394568 -7.380043 1.582378e-13 9.134437e-10
6      -10.23122 1.668505 -6.131971 8.679691e-10 1.473658e-06</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Sorting to show the highest log2FoldChange</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Is this up or down regulated?</span></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="fu">arrange</span>(df_de, <span class="fu">desc</span>(log2FoldChange)) <span class="sc">%&gt;%</span> <span class="fu">head</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                seqid sampleA sampleB baseMeanA baseMeanB  baseMean
1  comp180009_c0_seq2       Q       W 1258.9584 0.2022983 629.58036
2  comp175226_c0_seq1       Q       W  343.1854 0.0000000 171.59270
3 comp175480_c0_seq13       Q       W  260.8687 0.0000000 130.43435
4 comp173370_c0_seq14       Q       W  231.8312 0.0000000 115.91562
5  comp181016_c1_seq5       Q       W  132.8061 0.0000000  66.40304
6  comp164452_c0_seq1       Q       W 1013.5418 1.3837382 507.46276
  log2FoldChange    lfcSE     stat       pvalue         padj
1      12.075260 1.541190 7.835023 4.687565e-15 4.509906e-11
2      11.180387 1.548011 7.222422 5.106982e-13 2.456714e-09
3      10.785645 3.047203 3.539523 4.008503e-04 4.552410e-02
4      10.614133 1.565590 6.779636 1.204793e-11 4.346743e-08
5       9.810545 1.613226 6.081320 1.191973e-09 1.720196e-06
6       9.570938 1.743204 5.490428 4.009611e-08 3.403806e-05</code></pre>
</div>
</div>
<p>(2) Copy each <code>seqid</code> to search in <code>Elizabethae_Assembly_Trinity.fasta</code>. You can do that in <a href="https://usegalaxy.org/">Galaxy</a>:</p>
<ul>
<li><p>Use the tool <code>Filter FASTA on the headers and/or the sequences</code>.</p></li>
<li><p>Under <code>FASTA sequences</code>, choose <code>Elizabethae_Assembly_Trinity.fasta.</code></p></li>
<li><p>Under <strong><code>Criteria for filtering on the headers</code></strong>, choose <code>Regular expression….</code></p></li>
<li><p>Paste the seqid in the text box under <strong><code>Regular expression pattern</code></strong>.</p></li>
<li><p>(Alternatively, you can also download the assembly from Galaxy, open it in NotePad, and search.)</p></li>
</ul>
<p>(3) Copy that fasta sequence and do a <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi?LINK_LOC=blasthome&amp;PAGE_TYPE=BlastSearch&amp;PROGRAM=blastx">blastx</a> search against the refseq protein. What protein is this transcript most likely to be coding for?</p>
<p>Do a Google search using the name of the protein and the keyword “eusociality”. Is this protein related to eusociality?</p>
<hr>
</section>
<section id="rm-analysis" class="level2">
<h2 class="anchored" data-anchor-id="rm-analysis">RM analysis</h2>
<p>Let’s read <code>Elizabethae_Assembly_Trinity.fasta.out</code>, which is the output from Repeat Masker. Download the file <a href="https://github.com/solomonchak/R_Intro/raw/main/Elizabethae_Assembly_Trinity.fasta.out">here</a> and upload it to posit.cloud.</p>
<p>Repeat Masker was used to identify TEs from the transcriptome. Similar to Day 16, we will perform a few housing keeping fixes after loading the file.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>df_transcript_te <span class="ot">=</span> <span class="fu">read.table</span>(<span class="st">"Elizabethae_Assembly_Trinity.fasta.out"</span>, <span class="at">header =</span> F, <span class="at">skip =</span> <span class="dv">3</span>, <span class="at">fill =</span> T ,<span class="at">col.names =</span> <span class="fu">c</span>(<span class="st">"SW_score"</span>, <span class="st">"perc_div"</span>, <span class="st">"perc_del"</span>, <span class="st">"perc_inc"</span>, <span class="st">"seqid"</span>, <span class="st">"start"</span>, <span class="st">"end"</span>, <span class="st">"left"</span>, <span class="st">"strand"</span>,<span class="st">"repeat"</span>, <span class="st">"TE_class_family"</span>, <span class="st">"TE_start"</span>, <span class="st">"TE_end"</span>, <span class="st">"TE_left"</span>, <span class="st">"ID"</span>, <span class="st">"duplicate"</span>), )</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Some columns have numbers in () that needs to be removed.</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>df_transcript_te<span class="sc">$</span>left <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">"[()]"</span>,<span class="st">""</span>,df_transcript_te<span class="sc">$</span>left)</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>df_transcript_te<span class="sc">$</span>TE_start <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">"[()]"</span>,<span class="st">""</span>,df_transcript_te<span class="sc">$</span>TE_start)</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>df_transcript_te<span class="sc">$</span>TE_left <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">"[()]"</span>,<span class="st">""</span>,df_transcript_te<span class="sc">$</span>TE_left)</span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a><span class="co"># remove duplicates generated by Repeat Masker</span></span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>df_transcript_te <span class="ot">=</span> <span class="fu">filter</span>(df_transcript_te, <span class="sc">!</span>duplicate<span class="sc">==</span><span class="st">"*"</span>)</span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a><span class="co"># keep first entry for each seqid</span></span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a>df_transcript_te <span class="ot">=</span> <span class="fu">group_modify</span>((<span class="fu">group_by</span>(df_transcript_te, seqid)), <span class="sc">~</span>.x[<span class="dv">1</span>,])</span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a><span class="co"># get TE_class &amp; TE_family from TE_class_family</span></span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a>df_transcript_te<span class="sc">$</span>TE_class <span class="ot">=</span> <span class="fu">str_split</span>(df_transcript_te<span class="sc">$</span>TE_class_family, <span class="st">"/"</span>, <span class="at">simplify=</span> T)[,<span class="dv">1</span>] </span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a>df_transcript_te<span class="sc">$</span>TE_family <span class="ot">=</span> <span class="fu">str_split</span>(df_transcript_te<span class="sc">$</span>TE_class_family, <span class="st">"/"</span>, <span class="at">simplify=</span> T)[,<span class="dv">2</span>] </span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-18"><a href="#cb11-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Keep only the four major classes of TE (DNA, LINE, LTR and SINE).</span></span>
<span id="cb11-19"><a href="#cb11-19" aria-hidden="true" tabindex="-1"></a>df_transcript_te <span class="ot">=</span>  <span class="fu">filter</span>(df_transcript_te, TE_class<span class="sc">%in%</span><span class="fu">c</span>(<span class="st">"DNA"</span>, <span class="st">"LINE"</span>, <span class="st">"LTR"</span>, <span class="st">"SINE"</span>))</span>
<span id="cb11-20"><a href="#cb11-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-21"><a href="#cb11-21" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(df_transcript_te)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 18
# Groups:   seqid [6]
  seqid     SW_score perc_div perc_del perc_inc start   end left  strand repeat.
  &lt;chr&gt;        &lt;int&gt;    &lt;dbl&gt;    &lt;dbl&gt;    &lt;dbl&gt; &lt;int&gt; &lt;int&gt; &lt;chr&gt; &lt;chr&gt;  &lt;chr&gt;  
1 comp1002…      263     22.4      2.1      3.1   116   212 0     +      rnd-4_…
2 comp1003…      330     13.7      2.9      5.9   133   202 40    +      hAT-N2…
3 comp1007…      234     23.6      4.5      0     132   148 73    C      L2-8_N…
4 comp1008…      250     25.7      6.7      0      90   194 75    C      rnd-6_…
5 comp1008…      339      9.1      0        0       1    44 158   C      family…
6 comp1008…      288     17.1      4.8      1.2   172   183 71    +      L2-10_…
# ℹ 8 more variables: TE_class_family &lt;chr&gt;, TE_start &lt;chr&gt;, TE_end &lt;int&gt;,
#   TE_left &lt;chr&gt;, ID &lt;int&gt;, duplicate &lt;chr&gt;, TE_class &lt;chr&gt;, TE_family &lt;chr&gt;</code></pre>
</div>
</div>
<p>The <code>start</code> and <code>end</code> columns indicate where a TE is found in a transcript sequence. The name of the transcript is in <code>seqid</code>, which can later be matched with those in <code>df_de</code>.</p>
<p>Write some codes to make a new column <code>TE_length</code> as the absolute difference in <code>start</code> and <code>end</code> plus 1.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Your code</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="co"># checking the column TE_length should give you what's shown below.</span></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a><span class="co"># summary(df_transcript_te$TE_length)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
   11.0    54.0    74.0   114.9   121.0  2842.0 </code></pre>
</div>
</div>
<p>Let’s look at the TE distribution in the shrimp transcriptome. It’s very different from humans, which we looked at before.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot TE distribution</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(df_transcript_te, <span class="fu">aes</span>(TE_class)) <span class="sc">+</span> </span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>() <span class="sc">+</span> </span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>()  <span class="sc">+</span> </span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>(<span class="at">base_size=</span><span class="dv">14</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Project_Day2_files/figure-html/unnamed-chunk-13-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>We can also look at the distribution of perc_div across TE class. Note that this is only TEs that are in the transcriptome. It doesn’t reflect all the TEs that are in the shrimp genome.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot divergence facet</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(df_transcript_te, <span class="fu">aes</span>(perc_div, <span class="at">fill=</span>TE_class)) <span class="sc">+</span></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">colour=</span><span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>(<span class="at">base_size=</span><span class="dv">14</span>) <span class="sc">+</span></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>TE_class, <span class="at">scales =</span> <span class="st">"free"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
</div>
<div class="cell-output-display">
<p><img src="Project_Day2_files/figure-html/unnamed-chunk-14-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Finally, We can also check what’s the youngest element in each TE class, similar to your assignment in Day 16. We can do it with a ddply loop:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ddply</span>(df_transcript_te, <span class="st">"TE_class"</span>, <span class="cf">function</span>(i){</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ddply</span>(i, <span class="st">"TE_family"</span>, <span class="cf">function</span>(x){</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>(<span class="at">median =</span> <span class="fu">median</span>(x<span class="sc">$</span>perc_div))</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>    }) <span class="sc">%&gt;%</span> </span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(median) <span class="sc">%&gt;%</span></span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>  .[<span class="dv">1</span>,<span class="dv">1</span>]</span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>})</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  TE_class           V1
1      DNA TcMar-ISRm11
2     LINE     RTE-ORTE
3      LTR       Gypsy?
4     SINE    tRNA-Core</code></pre>
</div>
</div>
<p>Next week, we will do some demo analyses after combining the DE and RM data set. Then you’ll be assigned to work on similar analyses using one of the TE class.</p>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>
