<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.475">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Mohab Afghani">
<meta name="dcterms.date" content="2023-05-09">

<title>Final Project Report</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="Final_Project_files/libs/clipboard/clipboard.min.js"></script>
<script src="Final_Project_files/libs/quarto-html/quarto.js"></script>
<script src="Final_Project_files/libs/quarto-html/popper.min.js"></script>
<script src="Final_Project_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="Final_Project_files/libs/quarto-html/anchor.min.js"></script>
<link href="Final_Project_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="Final_Project_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="Final_Project_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="Final_Project_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="Final_Project_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body>

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction"><span class="toc-section-number">1</span>  Introduction</a></li>
  <li><a href="#methods-results" id="toc-methods-results" class="nav-link" data-scroll-target="#methods-results"><span class="toc-section-number">2</span>  Methods &amp; Results</a>
  <ul class="collapse">
  <li><a href="#differential-expressed-transcripts" id="toc-differential-expressed-transcripts" class="nav-link" data-scroll-target="#differential-expressed-transcripts"><span class="toc-section-number">2.1</span>  Differential expressed transcripts</a></li>
  <li><a href="#transcribed-transposable-elements" id="toc-transcribed-transposable-elements" class="nav-link" data-scroll-target="#transcribed-transposable-elements"><span class="toc-section-number">2.2</span>  Transcribed transposable elements</a></li>
  <li><a href="#are-there-more-tes-in-differentially-expressed-transcripts" id="toc-are-there-more-tes-in-differentially-expressed-transcripts" class="nav-link" data-scroll-target="#are-there-more-tes-in-differentially-expressed-transcripts"><span class="toc-section-number">2.3</span>  Are there more TEs in differentially expressed transcripts?</a></li>
  <li><a href="#does-te-transcripts-result-in-stronger-differential-expression" id="toc-does-te-transcripts-result-in-stronger-differential-expression" class="nav-link" data-scroll-target="#does-te-transcripts-result-in-stronger-differential-expression"><span class="toc-section-number">2.4</span>  Does TE transcripts result in stronger differential expression?</a></li>
  <li><a href="#does-te-length-have-an-effect-on-differential-expression" id="toc-does-te-length-have-an-effect-on-differential-expression" class="nav-link" data-scroll-target="#does-te-length-have-an-effect-on-differential-expression"><span class="toc-section-number">2.5</span>  Does TE Length have an effect on differential expression?</a></li>
  </ul></li>
  <li><a href="#conclusion" id="toc-conclusion" class="nav-link" data-scroll-target="#conclusion"><span class="toc-section-number">3</span>  Conclusion</a></li>
  </ul>
</nav>
</div>
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Final Project Report</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Mohab Afghani </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">May 9, 2023</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<section id="introduction" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Introduction</h1>
<p>Apart from the contribution of all transposable elements, we will also test whether LTR has a significant contribution to worker-queen differentiation. Furthermore, we will test to see if the transcripts that has TE may be more strongly differentially expressed in queens and if longer transposable elements are strongly diferentially expressed.</p>
<p>According to Grynberg et al.&nbsp;(2010), transposable elements (TEs) can be categorized into 2 separate classes, based on their dispersion mechanisms. Class I retrotransposons require an RNA intermediate in their transposition cycle and they synthesize a cDNA copy based on the RNA strand using reverse transcription <a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0029425#s1">(Grynberg et al.&nbsp;2010)</a>. Class II elements follow only an excision and insertion cycle.</p>
<p>Here, we hypothesize that LTRs in the TEs class are not contributing to the differentiation between queens and workers in eusocial shrimps. We expect to equal amounts of TEs in differentially expressed (DE) genes than in non-DE genes.</p>
</section>
<section id="methods-results" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Methods &amp; Results</h1>
<p>From the eusocial shrimp <em>Synalpheus elizabethae</em>, we obtained RNA-seq data from three queens and three workers. The transcriptome is assembled with Trinity (<a href="https://www.nature.com/articles/nprot.2013.084">Haas et al.&nbsp;2013</a>) and transposable elements were identified with Repeat Masker (<a href="https://currentprotocols.onlinelibrary.wiley.com/doi/10.1002/0471250953.bi0410s25">Tarailo-Graovac and Chen 2009</a>). We performed differential gene expression analysis using Galaxy (<a href="https://currentprotocols.onlinelibrary.wiley.com/doi/abs/10.1002/0471142727.mb1910s89">Blankenberg et al.&nbsp;2010</a>) and DESeq2 (<a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-014-0550-8?ref=https://githubhelp.com">Love et al.&nbsp;2014</a>). All statistical analyses were performed in R and detailed below.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"tidyr"</span>)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"dplyr"</span>)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"ggpubr"</span>)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"stringr"</span>)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"plyr"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="differential-expressed-transcripts" class="level2" data-number="2.1">
<h2 data-number="2.1" class="anchored" data-anchor-id="differential-expressed-transcripts"><span class="header-section-number">2.1</span> Differential expressed transcripts</h2>
<p>Among 29,833 transcripts that were shared among the six samples, 122 are up regulated and 141 are down regulated.</p>
<div class="cell">

</div>
</section>
<section id="transcribed-transposable-elements" class="level2" data-number="2.2">
<h2 data-number="2.2" class="anchored" data-anchor-id="transcribed-transposable-elements"><span class="header-section-number">2.2</span> Transcribed transposable elements</h2>
<p>From the transcriptome, 8,970 transcripts contained transposable elements. The more abundant TE class was DNA transposons, followed by LTR and LINE, with SINE being the least abundant.</p>
<div class="cell">

</div>
</section>
<section id="are-there-more-tes-in-differentially-expressed-transcripts" class="level2" data-number="2.3">
<h2 data-number="2.3" class="anchored" data-anchor-id="are-there-more-tes-in-differentially-expressed-transcripts"><span class="header-section-number">2.3</span> Are there more TEs in differentially expressed transcripts?</h2>
<p>If TEs are contributing to the evolution of the worker-queen differentiation, we expect to find more TEs in differentially expressed (DE) genes than in non-DE genes. Therefore, we performed a Chi-sqared test of independence on the binomial predictor variable DE (DE vs.&nbsp;NS, for differentially expressed vs.&nbsp;non-significant) against the response variable TE (TE vs.&nbsp;not, for having TE vs.&nbsp;not).</p>
<p>The results showed that the variables DE and TE are independent (Chi-squared = 2.160, d.f. = 1, p = 0.142). This means that the proportions of transcripts having TEs or not are not statistically different between transcripts that are and are not differentially expressed between queens and workers</p>
<div class="cell">

</div>
<p>Further, we tested and showed that TE class (DNA, LTR, LINE, and SINE) did not affect the variable DE (Chi-squared = 1.204, d.f. = 3, p = 0.752).</p>
<div class="cell">

</div>
</section>
<section id="does-te-transcripts-result-in-stronger-differential-expression" class="level2" data-number="2.4">
<h2 data-number="2.4" class="anchored" data-anchor-id="does-te-transcripts-result-in-stronger-differential-expression"><span class="header-section-number">2.4</span> Does TE transcripts result in stronger differential expression?</h2>
<p>If TEs are contributing to the evolution of the worker-queen differentiation, then it is believed that the transcripts that has TE may be more strongly differentially expressed in queens. In order to determine this, a two sample t-test is used and its results are analyzed so that we can test if our hypothesis is correct.</p>
<p>The results showed that the variables TE and log2FoldChange (DE) are dependent (t.test = -6.9077, df = 4547.2, p-value = 5.608e-12). This means that the transposable elements are more strongly differentially expressed in queens then workers, aiding in the evolution of worker-queen differentiation. Thus, our theory is supported by the results.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Predictor = TE (Y or N, categorical &amp; binomial), Response = log2FoldChange (continuous), use log(abs(log2FoldChange))</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="co"># use t-test </span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="fu">t.test</span>(<span class="fu">log</span>(<span class="fu">abs</span>(log2FoldChange))<span class="sc">~</span>TE, <span class="at">data =</span> df_tranTE) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
    Welch Two Sample t-test

data:  log(abs(log2FoldChange)) by TE
t = -6.9077, df = 4547.2, p-value = 5.608e-12
alternative hypothesis: true difference in means between group Not and group TE is not equal to 0
95 percent confidence interval:
 -0.1908548 -0.1064702
sample estimates:
mean in group Not  mean in group TE 
       -0.7018847        -0.5532222 </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># plot data</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> df_tranTE, <span class="fu">aes</span>(<span class="at">x =</span> TE, <span class="at">y =</span> <span class="fu">log</span>(<span class="fu">abs</span>(log2FoldChange)))) <span class="sc">+</span> </span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"TE"</span>, <span class="at">y =</span> <span class="st">"log(abs(log2FoldChange))"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Final_Project_files/figure-html/unnamed-chunk-7-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Now we filter out the TE class so that only the LTR TEs are shown and determine if our theory is still correct.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a table that filters out TE class with only LTR</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>df_tranTE_LTR <span class="ot">=</span> <span class="fu">filter</span>(df_tranTE, TE_class <span class="sc">==</span> <span class="st">"LTR"</span><span class="sc">|</span><span class="fu">is.na</span>(TE_class))</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Predictor = LTR(TE) (Y or N, categorical &amp; binomial)</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Response = log2FoldChange (continuous), use log(abs(log2FoldChange))</span></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a><span class="co"># use t-test </span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a><span class="fu">t.test</span>(<span class="fu">log</span>(<span class="fu">abs</span>(log2FoldChange))<span class="sc">~</span>TE, <span class="at">data =</span> df_tranTE_LTR) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
    Welch Two Sample t-test

data:  log(abs(log2FoldChange)) by TE
t = -1.786, df = 995.77, p-value = 0.07441
alternative hypothesis: true difference in means between group Not and group TE is not equal to 0
95 percent confidence interval:
 -0.156975174  0.007387353
sample estimates:
mean in group Not  mean in group TE 
       -0.7018847        -0.6270908 </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># plot data</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> df_tranTE_LTR, <span class="fu">aes</span>(<span class="at">x =</span> TE, <span class="at">y =</span> <span class="fu">log</span>(<span class="fu">abs</span>(log2FoldChange)))) <span class="sc">+</span> </span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"LTR"</span>, <span class="at">y =</span> <span class="st">"log(abs(log2FoldChange))"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Final_Project_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>The results showed that the variables LTR (TE) and log2FoldChange (DE) are independent (t.test = -1.786, df = 995.77, p-value = 0.07441 ). This means that the LTR class of TEs does not have an effect on differnetial expression or not are not statistically different between transcripts that are and are differentially expressed between queens and workers. This does not support our theory that TE may be more strongly differentially expressed in queens.</p>
</section>
<section id="does-te-length-have-an-effect-on-differential-expression" class="level2" data-number="2.5">
<h2 data-number="2.5" class="anchored" data-anchor-id="does-te-length-have-an-effect-on-differential-expression"><span class="header-section-number">2.5</span> Does TE Length have an effect on differential expression?</h2>
<p>If TEs are contributing to the evolution of the worker-queen differentiation, then it is believed that the transcripts that has longer TE may be more strongly diferentially expressed. In order to determine this, a linear regression test is used and its results are analyzed so that we can test if our hypothesis is correct.</p>
<p>The results showed that the variables TE_length and log2FoldChange (DE) are independent (F-statistic= 2.125 on 1 and 931 DF, p-value= 0.1452). This means that the transposable elements’ lengths do not have an effect on how strongly a TE is differentially expressed, giving us a slope and intercept very close to zero. Thus, our theory is not supported with this result.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Predictor = TE_length (continuous). Note that not all genes in df_tranTE has TE, so we’ll need to filter it to just rows with TEs. We should transform it with log(TE_length) to conform with normality.</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Response = log2FoldChange (continuous). Note again that log2FoldChange can be analyzed in two ways. We will use log(abs(log2FoldChange)) here.</span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Use linear regression </span></span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>lm_HA <span class="ot">=</span> <span class="fu">lm</span>(<span class="at">formula =</span> <span class="fu">log</span>(<span class="fu">abs</span>(log2FoldChange)) <span class="sc">~</span> <span class="fu">log</span>(TE_length), <span class="at">data =</span> df_tranTE)</span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>lm_HA</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = log(abs(log2FoldChange)) ~ log(TE_length), data = df_tranTE)

Coefficients:
   (Intercept)  log(TE_length)  
     -0.522573       -0.007008  </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot the data </span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> df_tranTE, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">log</span>(TE_length), <span class="at">y =</span> <span class="fu">log</span>(<span class="fu">abs</span>(log2FoldChange)))) <span class="sc">+</span></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method =</span> lm, <span class="at">linetype=</span> <span class="st">"dashed"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`geom_smooth()` using formula = 'y ~ x'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 26335 rows containing non-finite values (`stat_smooth()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 26335 rows containing missing values (`geom_point()`).</code></pre>
</div>
<div class="cell-output-display">
<p><img src="Final_Project_files/figure-html/unnamed-chunk-11-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Now we filter out the TE class so that only the LTR TEs are shown and determine if our theory is still correct.</p>
<p>Hypothesis- If the hypothesis is true (TEs are contributing to the evolution of the worker-queen differentiation), transcripts that has longer TE may be more strongly diferentially expressed.</p>
<p>The results showed that the variables TE length of LTR’s and log2FoldChange (DE) are also independent (linear regression = 2.125, D.F = 1 and 931, p-value: 0.1452). We fail to reject the null hypothesis because the p value &gt; 0.05. This means that the length of TEs in the LTR class does not have an effect on differential expression or not are statistically different between transcripts that are and are not differentially expressed between queens and workers.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Predictor = TE_length (continuous). Note that not all genes in df_tranTE has TE, so we’ll need to filter it to just rows with TEs. We should transform it with log(TE_length) to conform with normality.</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Response = log2FoldChange (continuous). Note again that log2FoldChange can be analyzed in two ways. We will use log(abs(log2FoldChange)) here.</span></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Use linear regression </span></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>lm_HA_LTR <span class="ot">=</span> <span class="fu">lm</span>(<span class="at">formula =</span> <span class="fu">log</span>(<span class="fu">abs</span>(log2FoldChange)) <span class="sc">~</span> <span class="fu">log</span>(TE_length), <span class="at">data =</span> df_tranTE_LTR)</span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>lm_HA_LTR</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = log(abs(log2FoldChange)) ~ log(TE_length), data = df_tranTE_LTR)

Coefficients:
   (Intercept)  log(TE_length)  
      -0.25215        -0.08625  </code></pre>
</div>
</div>
<p>Plot the Linear Regression</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot the data </span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> df_tranTE_LTR, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">log</span>(TE_length), <span class="at">y =</span> <span class="fu">log</span>(<span class="fu">abs</span>(log2FoldChange)))) <span class="sc">+</span></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method =</span> lm, <span class="at">linetype=</span> <span class="st">"dashed"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`geom_smooth()` using formula = 'y ~ x'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 26335 rows containing non-finite values (`stat_smooth()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 26335 rows containing missing values (`geom_point()`).</code></pre>
</div>
<div class="cell-output-display">
<p><img src="Final_Project_files/figure-html/unnamed-chunk-13-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Summary </span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(lm_HA_LTR)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = log(abs(log2FoldChange)) ~ log(TE_length), data = df_tranTE_LTR)

Residuals:
    Min      1Q  Median      3Q     Max 
-5.3419 -0.6897  0.1683  0.8012  2.8746 

Coefficients:
               Estimate Std. Error t value Pr(&gt;|t|)
(Intercept)    -0.25215    0.26047  -0.968    0.333
log(TE_length) -0.08625    0.05916  -1.458    0.145

Residual standard error: 1.257 on 931 degrees of freedom
  (26335 observations deleted due to missingness)
Multiple R-squared:  0.002277,  Adjusted R-squared:  0.001206 
F-statistic: 2.125 on 1 and 931 DF,  p-value: 0.1452</code></pre>
</div>
</div>
</section>
</section>
<section id="conclusion" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> Conclusion</h1>
<p>In summary, we found that transcripts that have TEs are not more differentially expressed. Also, the class of TE does not affect whether a transcribe is differentially expressed. There were unexpected results in our research where it was believed that the length of the TEs are not more strongly diferentially expressed. We also determined that TE transcripts do result in stronger differrential expression. However, when we filtered the TE class to just LTR, the results differed. It was determined that LTR does not have an effect on how strong a TE if differentially expressed. These results do not support the hypothesis that TEs are contributing to the evolution of the worker-queen differentiation.</p>
<p>Although the results did not support the conclusion, the dichotomous classification of transcripts into merely differentially expressed or not may conceal important pattern in the strength of differential expression. Further analysis should consider using a continuous variable based on the fold change of differential expression.</p>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>