<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.475">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Diandra Gordon">
<meta name="dcterms.date" content="2023-04-26">

<title>Final project</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="Diandra Gordon_files/libs/clipboard/clipboard.min.js"></script>
<script src="Diandra Gordon_files/libs/quarto-html/quarto.js"></script>
<script src="Diandra Gordon_files/libs/quarto-html/popper.min.js"></script>
<script src="Diandra Gordon_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="Diandra Gordon_files/libs/quarto-html/anchor.min.js"></script>
<link href="Diandra Gordon_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="Diandra Gordon_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="Diandra Gordon_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="Diandra Gordon_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="Diandra Gordon_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body>

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction"><span class="toc-section-number">1</span>  Introduction</a></li>
  <li><a href="#methods-results" id="toc-methods-results" class="nav-link" data-scroll-target="#methods-results"><span class="toc-section-number">2</span>  Methods &amp; Results</a>
  <ul class="collapse">
  <li><a href="#could-te-transcripts-be-strongly-deferentially-expressed" id="toc-could-te-transcripts-be-strongly-deferentially-expressed" class="nav-link" data-scroll-target="#could-te-transcripts-be-strongly-deferentially-expressed"><span class="toc-section-number">2.1</span>  Could TE transcripts be strongly deferentially expressed?</a></li>
  <li><a href="#does-te-length-have-an-effect-on-differential-expression" id="toc-does-te-length-have-an-effect-on-differential-expression" class="nav-link" data-scroll-target="#does-te-length-have-an-effect-on-differential-expression"><span class="toc-section-number">2.2</span>  Does TE length have an effect on differential expression?</a></li>
  </ul></li>
  <li><a href="#conclusion" id="toc-conclusion" class="nav-link" data-scroll-target="#conclusion"><span class="toc-section-number">3</span>  Conclusion</a></li>
  </ul>
</nav>
</div>
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Final project</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Diandra Gordon </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">April 26, 2023</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<section id="introduction" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Introduction</h1>
<p>See <a href="https://solomonchak.github.io/R_Intro/Project_Day1.html">here</a> for the background of the project.</p>
<p>Here, we hypothesize that TEs are contributing to the differentiation between queens and workers in eusocial shrimps, We expect to find more TEs in differentially expressed genes than in non-DE genes and transcripts that have longer TE may be more strongly deferentially expressed. Apart from the contribution of all TEs, we will also test whether DNA transposons have a significant contribution to worker-queen differentiation. DNA transposons are known to move from one genomic location to another by a cut-and-paste mechanism. DNA transposons are powerful forces of genetic change and have played a significant role in the evolution of many genomes (<a href="https://www.ingentaconnect.com/content/ben/cg/2010/00000011/00000002/art00004">Munoz-Lopez&amp;Garcia 2010</a>). They can be used to introduce foreign DNA into a genome, analyse the regulatory genome, study embryonic development, identify genes and pathways implicated in disease, and contribute to gene therapy.</p>
</section>
<section id="methods-results" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Methods &amp; Results</h1>
<p>From the eusocial shrimp <em>Synalpheus elizabethae</em>, we obtained RNA-seq data from three queens and three workers. The transcriptome is assembled with Trinity (<a href="https://www.nature.com/articles/nprot.2013.084">Haas et al.&nbsp;2013</a>) and transposable elements were identified with Repeat Masker (<a href="https://currentprotocols.onlinelibrary.wiley.com/doi/10.1002/0471250953.bi0410s25">Tarailo-Graovac and Chen 2009</a>). We performed differential gene expression analysis using Galaxy (<a href="https://currentprotocols.onlinelibrary.wiley.com/doi/abs/10.1002/0471142727.mb1910s89">Blankenberg et al.&nbsp;2010</a>) and DESeq2 (<a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-014-0550-8?ref=https://githubhelp.com">Love et al.&nbsp;2014</a>). All statistical analyses were performed in R and detailed below.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"tidyr"</span>)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"dplyr"</span>)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"ggpubr"</span>)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"stringr"</span>)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"plyr"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="could-te-transcripts-be-strongly-deferentially-expressed" class="level2" data-number="2.1">
<h2 data-number="2.1" class="anchored" data-anchor-id="could-te-transcripts-be-strongly-deferentially-expressed"><span class="header-section-number">2.1</span> Could TE transcripts be strongly deferentially expressed?</h2>
<div class="cell">

</div>
<div class="cell">

</div>
<p>Hypothesis: If the hypothesis is true, it’ll be considered that transcripts that have TE may be more strongly deferentially expressed in queens.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">t.test</span>(<span class="fu">log</span>(<span class="fu">abs</span>(log2FoldChange))<span class="sc">~</span>TE, <span class="at">data =</span> df_tranTE)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
    Welch Two Sample t-test

data:  log(abs(log2FoldChange)) by TE
t = -6.6975, df = 4541.1, p-value = 2.379e-11
alternative hypothesis: true difference in means between group Not and group TE is not equal to 0
95 percent confidence interval:
 -0.1872263 -0.1024362
sample estimates:
mean in group Not  mean in group TE 
       -0.7029916        -0.5581603 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot your data </span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> df_tranTE, <span class="fu">aes</span>(<span class="at">x =</span> TE, <span class="at">y =</span> <span class="fu">log</span>(<span class="fu">abs</span>(log2FoldChange)))) <span class="sc">+</span> </span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Diandra-Gordon_files/figure-html/unnamed-chunk-4-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Summary, we rejected the null meaning the hypothesis is true that TE has a significant of effect on dealing with differential expression</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># create a table filtering out TE class with DNA </span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>df_tranTE_DNA <span class="ot">=</span> <span class="fu">filter</span>(df_tranTE, TE_class<span class="sc">==</span><span class="st">"DNA"</span><span class="sc">|</span> <span class="fu">is.na</span>(TE_class))</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a><span class="co"># use t.test</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a><span class="fu">t.test</span>(<span class="fu">log</span>(<span class="fu">abs</span>(log2FoldChange))<span class="sc">~</span>TE, <span class="at">data =</span> df_tranTE_DNA)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
    Welch Two Sample t-test

data:  log(abs(log2FoldChange)) by TE
t = -3.8127, df = 1606.9, p-value = 0.0001426
alternative hypothesis: true difference in means between group Not and group TE is not equal to 0
95 percent confidence interval:
 -0.18244839 -0.05849529
sample estimates:
mean in group Not  mean in group TE 
       -0.7029916        -0.5825198 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co">#What are the predictor &amp; response variables?</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="co">#Predictor = TE (Y or N, categorical &amp; binomial)</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="co">#Response =  log2FoldChange(continuous) = log(abs(log2FoldChange))</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Summary, we rejected the null meaning the hypothesis is true, DNA has a significant effect on dealing with differential expression. (t= = -6.6975, df = 4541.1, p = 2.379e-11).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot your data </span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> df_tranTE_DNA, <span class="fu">aes</span>(<span class="at">x =</span> TE, <span class="at">y =</span> <span class="fu">log</span>(<span class="fu">abs</span>(log2FoldChange)))) <span class="sc">+</span> </span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"DNA"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Diandra-Gordon_files/figure-html/unnamed-chunk-6-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="does-te-length-have-an-effect-on-differential-expression" class="level2" data-number="2.2">
<h2 data-number="2.2" class="anchored" data-anchor-id="does-te-length-have-an-effect-on-differential-expression"><span class="header-section-number">2.2</span> Does TE length have an effect on differential expression?</h2>
<p>Predictions: If the hypothesis is true it’ll be considered that, transcripts that have longer TE may be more strongly deferentially expressed.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># # use liner regression</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>lm_HA <span class="ot">=</span> <span class="fu">lm</span>(<span class="at">formula =</span> <span class="fu">log</span>(<span class="fu">abs</span>(log2FoldChange)) <span class="sc">~</span> <span class="fu">log</span>(TE_length), <span class="at">data =</span> df_tranTE)</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(lm_HA)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = log(abs(log2FoldChange)) ~ log(TE_length), data = df_tranTE)

Residuals:
    Min      1Q  Median      3Q     Max 
-8.1856 -0.6526  0.1265  0.7574  3.8145 

Coefficients:
                Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)    -0.539360   0.120689  -4.469 8.11e-06 ***
log(TE_length) -0.004298   0.027203  -0.158    0.874    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 1.198 on 3499 degrees of freedom
  (26332 observations deleted due to missingness)
Multiple R-squared:  7.136e-06, Adjusted R-squared:  -0.0002787 
F-statistic: 0.02497 on 1 and 3499 DF,  p-value: 0.8745</code></pre>
</div>
</div>
<p>Summary: The null hypothesis failed to reject meaning that the length of TE has no effect on differential expression and there are is significant relationship between the two variables (log2foldchange) and (TE_length).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co"># create a table filtering out df_tranTE with DNA</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>lm_HA_DNA <span class="ot">=</span> <span class="fu">lm</span>(<span class="at">formula =</span> <span class="fu">log</span>(<span class="fu">abs</span>(log2FoldChange)) <span class="sc">~</span> <span class="fu">log</span>(TE_length), <span class="at">data =</span> df_tranTE_DNA)</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(lm_HA_DNA)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = log(abs(log2FoldChange)) ~ log(TE_length), data = df_tranTE_DNA)

Residuals:
    Min      1Q  Median      3Q     Max 
-7.3724 -0.6296  0.1058  0.7590  3.0876 

Coefficients:
               Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)    -0.72527    0.19490  -3.721 0.000206 ***
log(TE_length)  0.03261    0.04396   0.742 0.458389    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 1.159 on 1426 degrees of freedom
  (26332 observations deleted due to missingness)
Multiple R-squared:  0.0003856, Adjusted R-squared:  -0.0003154 
F-statistic: 0.5501 on 1 and 1426 DF,  p-value: 0.4584</code></pre>
</div>
</div>
<p>Summary, the null hypothesis failed to reject meaning that the length of TE has no effect on differential expression (DE). (F= 0.05501, df = 1 and 1426, p =0.4584)</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot your data</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> df_tranTE_DNA, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">log</span>(TE_length), <span class="at">y =</span> <span class="fu">log</span>(<span class="fu">abs</span>(log2FoldChange)))) <span class="sc">+</span></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method =</span> lm)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`geom_smooth()` using formula = 'y ~ x'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 26332 rows containing non-finite values (`stat_smooth()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 26332 rows containing missing values (`geom_point()`).</code></pre>
</div>
<div class="cell-output-display">
<p><img src="Diandra-Gordon_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
</section>
<section id="conclusion" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> Conclusion</h1>
<p>In summary, we found that transcripts that have TEs are more deferentially expressed. Also, the class of DNA does effect whether a transcribe is deferentially expressed. These results support the hypothesis that TEs are contributing to the evolution of the worker-queen differentiation.</p>
<p>Although the results did not support the conclusion, the dichotomous classification of transcripts into merely deferentially expressed or not may conceal important pattern in the strength of differential expression. Further analysis should consider using a continuous variable based on the fold change of differential expression.</p>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>